crates/forge/tests/integration/precalculate_address.rs

```diff
@@ -0,0 +1,64 @@
+use crate::integration::common::corelib::{corelib_path, predeployed_contracts};
+use crate::integration::common::runner::Contract;
+use crate::{assert_passed, test_case};
+use camino::Utf8PathBuf;
+use forge::run;
+use indoc::indoc;
+use std::path::Path;
+
+#[test]
+fn test_get_contract_precalculate_address() {
+    let test = test_case!(
+        indoc!(
+            r#"
+        use result::ResultTrait;
+        use cheatcodes::{ declare, ContractClass, ContractClassTrait };
+        use array::ArrayTrait;
+        use traits::Into;
+        use traits::TryInto;
+        use starknet::ContractAddressIntoFelt252;
+
+        #[starknet::interface]
+        trait IHelloStarknet<TContractState> {
+            fn increase_balance(ref self: TContractState, amount: felt252);
+            fn get_balance(self: @TContractState) -> felt252;
+            fn do_a_panic(self: @TContractState);
+            fn do_a_panic_with(self: @TContractState, panic_data: Array<felt252>);
+        }
```

Comment:
You don't need to copy the interface here since we do not use it in this test anyways